src/script/zcash_script.cpp

```diff
@@ -69,6 +69,21 @@ struct ECCryptoClosure
 };
 
 ECCryptoClosure instance_of_eccryptoclosure;
+
+/// Copy of GetLegacySigOpCount from main.cpp commit c4b2ef7c4
+unsigned int GetLegacySigOpCount(const CTransaction& tx)
+{
+    unsigned int nSigOps = 0;
+    for (const CTxIn& txin : tx.vin)
+    {
+        nSigOps += txin.scriptSig.GetSigOpCount(false);
+    }
+    for (const CTxOut& txout : tx.vout)
+    {
+        nSigOps += txout.scriptPubKey.GetSigOpCount(false);
+    }
+    return nSigOps;
+}
```

Comment:
I didn't want to link to `main.cpp`, because it has a lot of dependencies.
But I wasn't sure if you wanted to move legacy Bitcoin code.

I'm happy to remove the copy in `main.cpp` and put it in `zcash_script.h` or `script.h` instead.

Reply:
I'm fine with having this duplicated here for now. Upstream refactors it into `src/consensus/tx_verify.{cpp,h}`, so once we backport that refactor we can eliminate the duplication.

Reply:
I documented how to get rid of the duplicate in a comment.